File: frontend/src/models/driver/Driver.ts
export interface DriverSummary {
    id: number
    firstName: string
    lastName: string
}

export interface DriverDetails extends DriverSummary {
    login: string
    phoneNumber: string

    annualSalary: number
    monthlyWorkingHours: number

    tractorId: number | null
    tractorRegistration: string | null
    semiTrailerId: number | null
    semiTrailerRegistration: string | null
    costType: DriverCostType
    driverType: DriverType
}

export type DriverCostType = "HOURLY" | "FIXED"
export type DriverType = "LONG_HAUL" | "SHORT_DISTANCE" | "REGIONAL"

export interface CreateDriverRequest {
    firstName: string | null
    lastName: string | null
    phoneNumber: string | null
    annualSalary: number | null
    monthlyWorkingHours: number | null
    tractorId: number | null
    semiTrailerId: number | null
    costType: DriverCostType | null
    driverType: DriverType | null
}

export interface UpdateDriverRequest {
    firstName: string | null
    lastName: string | null
    phoneNumber: string | null
    annualSalary: number | null
    monthlyWorkingHours: number | null
    tractorId: number | null
    semiTrailerId: number | null
    costType: DriverCostType | null
    driverType: DriverType | null
}

export interface DriverFormData {
    firstName: string | null
    lastName: string | null
    phoneNumber: string | null
    annualSalary: number | null
    monthlyWorkingHours: number | null
    tractorId: number | null
    semiTrailerId: number | null
    costType: DriverCostType | null
    driverType: DriverType | null
}

File: backend/src/main/java/com/optiroute/backend/entity/driver/Driver.java
package com.optiroute.backend.entity.driver;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

import com.optiroute.backend.entity.EntityUtils;
import com.optiroute.backend.entity.vehicle.SemiTrailer;
import com.optiroute.backend.entity.vehicle.Tractor;

import com.optiroute.backend.type.driver.DriverCostType;
import com.optiroute.backend.type.driver.DriverType;

@Entity
@Table(name = "driver")
@Getter
@Setter
public class Driver extends EntityUtils {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false, unique = true)
    private String login;

    private String phoneNumber;

    @Column(nullable = false)
    private Boolean isActive = true;

    @Column(name = "annual_salary", precision = 10, scale = 2)
    private BigDecimal annualSalary;

    @Column(name = "monthly_working_hours", precision = 6, scale = 2)
    private BigDecimal monthlyWorkingHours;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tractor_id")
    private Tractor tractor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "semi_trailer_id")
    private SemiTrailer semiTrailer;

    @Enumerated(EnumType.STRING)
    @Column(name = "cost_type", nullable = false, length = 30)
    private DriverCostType costType;

    @Enumerated(EnumType.STRING)
    @Column(name = "driver_type", nullable = false, length = 30)
    private DriverType driverType;
}